impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java

```diff
@@ -205,7 +205,7 @@ else if (annotation instanceof Version version) {
             }
             else if (annotation instanceof SaslMechanism mechanism) {
                 useSasl = true;
-                builder.saslMechanism(Optional.ofNullable(mechanism.value()).orElse(PLAIN_SASL_MECHANISM_NAME));
```

Comment:
that's the unnecessary defaulting - it was only ever null when coming from the test proxy.